llama_index/storage/docstore/keyval_docstore.py

```diff
@@ -101,25 +122,91 @@ def get_document(
             raise_error (bool): raise error if doc_id not found
 
         """
-        json = self._kvstore.get(doc_id, collection=self._collection)
+        json = self._kvstore.get(doc_id, collection=self._node_collection)
         if json is None:
             if raise_error:
                 raise ValueError(f"doc_id {doc_id} not found.")
             else:
                 return None
         return json_to_doc(json)
 
+    def get_ref_doc_info(self, ref_doc_id: str) -> Optional[RefDocInfo]:
+        """Get the RefDocInfo for a given ref_doc_id."""
+        ref_doc_info = self._kvstore.get(
+            ref_doc_id, collection=self._ref_doc_collection
+        )
+        if not ref_doc_info:
+            return None
+        return RefDocInfo(**ref_doc_info)
+
+    def get_all_ref_doc_info(self) -> Optional[Dict[str, RefDocInfo]]:
+        """Get a mapping of ref_doc_id -> RefDocInfo for all ingested documents."""
+        ref_doc_infos = self._kvstore.get_all(collection=self._ref_doc_collection)
+        if ref_doc_infos is None:
+            return None
+        return {key: RefDocInfo(**val) for key, val in ref_doc_infos.items()}
+
+    def ref_doc_exists(self, ref_doc_id: str) -> bool:
```

Comment:
Is this a bit refundant with `document_exists`?

Reply:
Yes you are right. I was thinking about the case where document are actually inserted.